### PR TITLE
Add labels for repo and commit to packages

### PR DESCRIPTION
**Pull request by justincormack**
- use image spec metadata https://github.com/opencontainers/image-spec/blob/master/annotations.md
- omit commit if dirty

Signed-off-by: Justin Cormack <justin.cormack@docker.com>

fix https://github.com/moby/tool/issues/20

Initial commit for discussion before rebuilding everything...

mama-baby-bear-animals-36003651-500-350

### CODE REVIEW

**Comment by ijc** [comment]
https://github.com/opencontainers/image-spec/issues/658 was trying to address this from the image spec end. I'm not sure what the result was. The table in https://github.com/opencontainers/image-spec/issues/658#issuecomment-300336575 doesn't seem to have been implemented since [`annotations.md`](https://github.com/opencontainers/image-spec/blob/master/annotations.md) doesn't seem to cover everything listed there, in particular `source` and `revision` (corresponding to `vcs-url` and `vcs-ref` you've used here) seem to be missing.

File: pkg/package.mk [hunk 1]
```diff
@@ -13,6 +13,17 @@ endif
 
 TAG:=$(ORG)/$(IMAGE):$(HASH)$(DIRTY)
 
+REPO?=https://github.com/linuxkit/linuxkit
+ifneq ($(REPO),)
+REPO_LABEL=--label org.label-schema.vcs-url=$(REPO)
+endif
+REPO_DIRTY:=$(shell git update-index -q --refresh && git diff-index --quiet HEAD -- || echo "-dirty")
```
Comment (rn) [inline pkg/package.mk]: Doesn't the same logic we use for `DIRTY` apply here, ie we don't care if the repo is dirty as long as the directory is clean? It seems odd to me to apply tow different notions of dirty

**Comment by ijc** [comment]
Code LGTM. Possibly you could take `HEAD` if the subtree is clean rather than worrying about the whole repo, but the way you have it is equally fine.

**Comment by justincormack** [comment]
Yes, maybe will be consistent and use the same `DIRTY`

**Comment by justincormack** [comment]
Ok, changed to use the same `DIRTY`

**Comment by justincormack** [comment]
@ijc I think we can potentially add things to image spec metadata as well, but they are not very accessible compared to Docker image metadata at present, eg in terms of tooling, and we are generating Docker images not image spec images for now anyway.

**Comment by justincormack** [comment]
we could use `org.opencontainers.image.source` and `org.opencontainers.image.revision` as the tags instead see https://github.com/opencontainers/image-spec/blob/master/annotations.md - maybe better

**Comment by ijc** [comment]
Oh, those were the ones I claimed weren't specified in my comment above, but it turns out my local tree was just very out of date.

**Comment by justincormack** [comment]
Updated to use image spec labels as they specifically say they obsolete label schema

**Comment by ijc** [comment]
LGTM (maybe `LABEL_OPT` for consistency with `NET_OPT` but that's just pedantry on my behalf)

**Comment by justincormack** [comment]
Added to kernel build

**Comment by justincormack** [comment]
@ijc changed to `LABELS` in the interests of pedantry; it doesn't feel like an `OPT`

**Comment by justincormack** [comment]
@rn yes, will add a rebuild once we are agreed on everything here

**Comment by rn** [comment]
looks good to me, though, as we discussed, ideally we'd also include `package.mk` into the hash somehow, but that can be a separate effort

**Comment by justincormack** [comment]
I am updating the images, but as their hashes will not change this can be merged once CI is green.

You can see labels if you re-pull and do `docker inspect linuxkit/acpid:1966310cb75e28ffc668863a6577ee991327f918`

**Comment by justincormack** [comment]
Going to leave the image updates (some done) as the base image needs updating anyway, so will re-generate with labels anyway today/tomorrow.

**Pull request merged**